propagator/README.md

```diff
@@ -0,0 +1,51 @@
+# OpenTelemetry Google Cloud Trace Propagators
+
+[![Docs](https://godoc.org/github.com/GoogleCloudPlatform/opentelemetry-operations-go/propagator?status.svg)](https://pkg.go.dev/github.com/GoogleCloudPlatform/opentelemetry-operations-go/propagator)
+
+This package contains Trace Context Propagators for use with [Google Cloud
+Trace](https://cloud.google.com/trace) that make it compatible with
+[OpenTelemetry](http://opentelemetry.io). 
+
+There are two available propagators in this package:
+
+### `CloudTraceOneWayPropagator`
+
+The `CloudTraceOneWayPropagator` reads the `X-Cloud-Trace-Context` header for
+trace and span IDs, and writes this data into the `traceparent` header.
+
```

Comment:
Can we add a recommended usage golang snippet?  Something like:

```golang
propagation.NewCompositeTextMapPropagator(propagator.CloudTraceOneWayPropagator{}, propagation.TraceContext{}, propagation.Baggage{})
```

We want to make sure traceparent takes precedence over X-Cloud-Trace-Context when reading.

Reply:
Do we want to force users to create a composite propagator for Trace Context?
Multiple propagators using the same key in the TextMap can lead to silent failures and, IMO, should be avoided where possible.

From a DevEx perspective, a single Propagator would let us have explicit precedence (and tests to prove it!), in addition to being less prone to user error (e.g. by getting the order wrong). (this approach requires an embedded TraceContext propagator). Usage:
`propagation.NewCompositeTextMapPropagator(gcpprop.CloudTraceOneWayPropagator{}, propagation.Baggage{})` (if we go this way, i'd suggest we rename to `CloudTraceToTraceContextPropagator` or similar)

If we want to take the opposite approach, and make a single-purpose propagator, perhaps we should avoid having `Inject()` do anything, and instead rely on the user specifying their own propagator. DevEx on this would be more error-prone, since the order of propagators matters. Usage:
`propagation.NewCompositeTextMapPropagator(gcpprop.CloudTraceOneWayPropagator{}, propagation.TraceContext{}, propagation.Baggage{})`

Either way, the code needs to change, but we should be explicit about the choice. 😄 What do you think?

Reply:
For the first case, we would likely also want to call `propagation.TraceContext.Extract()` from as part of Extract as well.

The order of propagators only matters if both are present, which should be rare. There could be a case in which the user wants to prefer XCT (e.g. if they want to be a child of a GCLB span in GAE).  It would be more flexible to have it separate as well if someone wanted to use it with a different "base" propagator (e.g. jaeger) for some reason.

@aabmass any thoughts?

Reply:
I've chosen to go for the simpler DevEx, and made the Propagator read both headers, but prioritize `traceparent`.
This is a tradeoff with flexibility, as you mention, but helps users converge on the w3c standard, which seems like an acceptable compromise.
I also refactored the tests so that the priority of trace headers is clear.